src/Type/ArrayType.php

```diff
@@ -143,6 +143,16 @@ public function getIterableValueType(): Type
 		return $this->getItemType();
 	}
 
+	public function isOffsetAccesible(): TrinaryLogic
+	{
+		return TrinaryLogic::createYes();
+	}
+
+	public function getOffsetValueType(): Type
+	{
+		return $this->getIterableValueType();
```

Comment:
You should just return `$this->itemType`.
Returning `getIterableValueType` is very odd from semantic point of view because it means that value returned by offset access *depends* on value returned by iteration. Which is not the case.

Reply:
fixed